### ISSUE TITLE
DateTime field edit type

### ISSUE THREAD

**Issue by VieriNucci** [opengisch/QField#9]
I have installed QField 0.3.2 on a Samsung Galaxy Note 8.0. I have a project with a shapefile layer. One of the fields, called "Time" is defined as Date (QDAte) and widget is set to display "yyyy-MM-dd  HH:mm:ss", two other fields are called respectively Start_Date and End_Date, are defined as Date (QDate) and widget is set to display "yyyy-MM-dd" only.
However all fields do not automatically update when opened in QField editor, so that theoretically I'd have to insert all times and dates manually. On the other hand in QGIS for desktop everything is working fine.
When I use a spatialite layer the behavior is the same.

It may be useful to say that if a field was edited in QGIS (2.8.1), when opened in QField it displays as 2015-03-25T00:00:00, even if time was correct in QGIS. May be there a problem when copying the project file? Should I export projects and shapefiles differently, instead of just copying and pasting?

**Comment by m-kuhn** [comment]
Background:
One of the main differences of QField in comparison to QGIS is, that it uses [QML](http://doc.qt.io/qt-5/qmlapplications.html) instead of [QWidgets](http://doc.qt.io/qt-4.8/qt-gui-concepts.html).
As the edit widgets are implemented as QWidgets, they need to be re-done in QML style. So far we have ported only the TextEdit and the CheckBox widgets. All not yet ported field editors fallback to TextEdit. Hence no default values and goodies specific to these yet.

Long story short: The DateTime widget needs to be ported to QML.
